config/initializers/03_sidekiq.rb

```diff
@@ -11,9 +11,18 @@
     config.redis = redis_config
 
     config.death_handlers << ->(job, original_exception) do
-      if original_exception.is_a?(Pender::Exception::RetryLater)
-        limit_hit_exception = Pender::Exception::RetryLimitHit.new(original_exception)
-        PenderSentry.notify(limit_hit_exception, { job: job, original_exception: original_exception.cause.inspect })
+      case original_exception
+      when Pender::Exception::RetryLater
+        if original_exception.message.include?("Too Many Requests")
+          rate_limit_exception = Pender::Exception::RateLimitExceeded.new(original_exception)
+          PenderSentry.notify(rate_limit_exception, { job: job, original_exception: original_exception.cause.inspect })
+        else
+          limit_hit_exception = Pender::Exception::RetryLimitHit.new(original_exception)
+          PenderSentry.notify(limit_hit_exception, { job: job, original_exception: original_exception.cause.inspect })
```

Comment:
This line is the same as line 18. We can use the conditional to just set the exception variable, but have a single line for the Sentry notification.

Reply:
I added an exception variable here.